### PR TITLE
Add API to block network requests (Xamarin)

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [ ] ~Has `CHANGELOG.md` been updated?~
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests if this modifies the Windows code?~
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Has at least one reviewer tested the PR?

## Description

Add API to block network requests.

## Related PRs or issues

[AB#85223](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/85223)

### CODE REVIEW

**Review by russelarms — APPROVED** [review]

File: SDK/AppCenter/Microsoft.AppCenter.Shared/AppCenter.cs [hunk 1]
```diff
@@ -100,6 +100,15 @@ public static LogLevel LogLevel
             set => PlatformLogLevel = value;
         }
 
+        /// <summary>
+        ///     Allow or disallow network requests.
+        /// </summary>
+        public static bool NetworkRequestsAllowed
```
Comment (russelarms) [inline SDK/AppCenter/Microsoft.AppCenter.Shared/AppCenter.cs:106]: Why not `IsNetworkRequestsAllowed`?

**Review by DmitriyKirakosyan — APPROVED** [review]

File: Apps/Contoso.Android.Puppet/ModulePages/AppCenterFragment.cs [hunk 2]
```diff
@@ -54,6 +55,8 @@ public override void OnViewCreated(View view, Bundle savedInstanceState)
 
             // Find views.
             AppCenterEnabledSwitch = view.FindViewById(Resource.Id.enabled_app_center) as Switch;
+            AppcenterNetworkRequestsAllowedSwitch = view.FindViewById(Resource.Id.appcenter_network_requests_allowed) as Switch;
+
```
Comment (MatkovIvan) [inline Apps/Contoso.Android.Puppet/ModulePages/AppCenterFragment.cs:59]: comment-if-empty-line 😄

File: Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/App.xaml.cs [hunk 3]
```diff
@@ -50,6 +50,10 @@ protected override void OnStart()
                 AppCenter.LogLevel = LogLevel.Verbose;
                 AppCenterLog.Info(LogTag, "AppCenter.LogLevel=" + AppCenter.LogLevel);
                 AppCenterLog.Info(LogTag, "AppCenter.Configured=" + AppCenter.Configured);
+                if (Current.Properties.ContainsKey(Constants.AllowNetworkRequests) && Current.Properties[Constants.AllowNetworkRequests] is bool isAllow)
+                {
+                    AppCenter.IsNetworkRequestsAllowed = isAllow;
```
Comment (MatkovIvan) [inline Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/App.xaml.cs:55]: I'd rename `isAllow` variable, this name does not really explain what it is

File: Apps/Contoso.iOS.Puppet/AppDelegate.cs [hunk 4]
```diff
@@ -26,6 +26,13 @@ public override UIWindow Window
 
         public override bool FinishedLaunching(UIApplication application, NSDictionary launchOptions)
         {
+            // Init user default values.
+            var keys = new object[] { Constants.AllowNetworkRequests };
+            var values = new object[] { true };
+            NSUserDefaults userDefaults = new NSUserDefaults();
```
Comment (MatkovIvan) [inline Apps/Contoso.iOS.Puppet/AppDelegate.cs:32]: `NSUserDefaults` -> `var`

Comment (MatkovIvan) [inline Apps/Contoso.iOS.Puppet/AppDelegate.cs:32]: Why not `NSUserDefaults.StandardUserDefaults`?

File: SDK/AppCenter/Microsoft.AppCenter.Android/AppCenter.cs [hunk 5]
```diff
@@ -83,6 +83,18 @@ static LogLevel PlatformLogLevel
             }
         }
 
+        static bool PlatformIsNetworkRequestsAllowed
+        {
```
Comment (MatkovIvan) [inline SDK/AppCenter/Microsoft.AppCenter.Android/AppCenter.cs:87]:
```csharp
{
    get => AndroidAppCenter.NetworkRequestsAllowed;
    set => AndroidAppCenter.NetworkRequestsAllowed = value;
}
```

File: SDK/AppCenter/Microsoft.AppCenter.iOS/AppCenter.cs [hunk 6]
```diff
@@ -79,6 +79,18 @@ static LogLevel PlatformLogLevel
             }
         }
 
+        static bool PlatformIsNetworkRequestsAllowed
+        {
```
Comment (MatkovIvan) [inline SDK/AppCenter/Microsoft.AppCenter.iOS/AppCenter.cs:83]: Same

**Pull request merged**